### PR TITLE
[qt] move SendCoinsRecipient to its own file

### PR DESCRIPTION
**Pull request by Sjors**
I plan to add some instance methods to `SendCoinsRecipient` in an upcoming PR, so this might be a good time to give it its own file.

### CODE REVIEW

File: src/qt/walletmodeltransaction.cpp [hunk 1]
```diff
@@ -7,6 +7,8 @@
 #include <policy/policy.h>
 #include <wallet/wallet.h>
 
+class SendCoinsRecipient;
```
Comment (promag) [inline src/qt/walletmodeltransaction.cpp]: Remove?

**Review by promag — COMMENTED** [review]
IMO if it's really "necessary" to move to a separate file then that can happen in the PR that takes advantage of it.

Build job failed with
```
Making check in src
make[1]: Entering directory `/home/travis/build/bitcoin/bitcoin/build/bitcoin-x86_64-unknown-linux-gnu/src'
make[2]: Entering directory `/home/travis/build/bitcoin/bitcoin/build/bitcoin-x86_64-unknown-linux-gnu/src'
  OBJCXXLD qt/bitcoin-qt
  CXXLD    qt/test/test_bitcoin-qt
qt/libbitcoinqt.a(qt_libbitcoinqt_a-guiutil.o): In function `GUIUtil::parseBitcoinURI(QUrl const&, SendCoinsRecipient*)':
guiutil.cpp:(.text+0x557f): undefined reference to `SendCoinsRecipient::SendCoinsRecipient()'
qt/test/qt_test_test_bitcoin_qt-uritests.o: In function `URITests::uriTests()':
uritests.cpp:(.text+0x33): undefined reference to `SendCoinsRecipient::SendCoinsRecipient()'
collect2: error: ld returned 1 exit status
```

File: src/qt/walletmodeltransaction.cpp (hunk 1 above)
Comment (Sjors) [reply src/qt/walletmodeltransaction.cpp]: Removed.

**Comment by Sjors** [comment]
The Travis instance that builds QT without the wallet broke, so I added a bunch of `#ifdef ENABLE_WALLET` to `guiutil.cpp`, for example around `parseBitcoinURI`. I should probably do that in a separate commit.

> IMO if it's really "necessary" to move to a separate file then that can happen in the PR that takes advantage of it.

I know there's a (implicit?) policy against refactoring for the sake of refactoring. On the other hand, I like to keep my PR's short because I think it makes review easier. I can rebase my upcoming PR on top of this if people prefer that. Or we can hold off merging this until that PR is ready and has a good chance of getting merged.

**Comment by Sjors** [comment]
Not sure what to do about these `has no symbols` warnings:

```
  AR       qt/libbitcoinqt.a
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: qt/libbitcoinqt.a(qt_libbitcoinqt_a-moc_sendcoinsrecipient.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: qt/libbitcoinqt.a(qt_libbitcoinqt_a-moc_sendcoinsrecipient.o) has no symbols
  OBJCXXLD qt/bitcoin-qt
```

**Comment by MarcoFalke** [comment]
Needs rebase if still relevant

File: src/qt/openuridialog.cpp [hunk 2]
```diff
@@ -32,6 +32,7 @@ QString OpenURIDialog::getURI()
 
 void OpenURIDialog::accept()
 {
+#ifdef ENABLE_WALLET
```
Comment (jonasschnelli) [inline src/qt/openuridialog.cpp]: Is this #ifdef necessary? IMO `openuridialog.cpp/.h` is only in `BITCOIN_QT_WALLET_CPP`.

**Comment by jonasschnelli** [comment]
Concept ACK... but you should compile the new class always (since its not really depending on wallet classes)... also, avoid additional #ifdefs (IMO they are the last resort for selective features).

**Comment by fanquake** [comment]
Going to close for now. Needs a rebase. I also somewhat agree that a change like this can be incorporated into the first PR that will be making use of it.

**Comment by Sjors** [comment]
Ok, it also looks I'm going for a different approach for RBF than what I had in mind back then.

**Pull request closed without merging**